### PR TITLE
Explain nonpayable

### LINKED ISSUE #8736: [Documentation] Add clarification that "nonpayable" abi state does not correspond to any solidity state mutability keyword unlike other mutability states that map one-to-one
## Description

Since `nonpayable` mutability is not parseable e.g.,

```
contract I {
        function f() external virtual nonpayable {

        }
}
```

we need to remove reference to it here: https://solidity.readthedocs.io/en/v0.6.6/abi-spec.html?highlight=nonpayable#json

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/8736

### CODE REVIEW

**Review by axic — APPROVED** [review]

File: docs/abi-spec.rst [hunk 1]
```diff
@@ -507,13 +507,17 @@ A function description is a JSON object with the fields:
...
 
 Constructor and fallback function never have ``name`` or ``outputs``. Fallback function doesn't have ``inputs`` either.
 
 .. note::
     Sending non-zero Ether to non-payable function will revert the transaction.
 
+.. note::
+    The state mutability ``nonpayable`` is reflected in Solidity by not specifying
+    a state mutability modifier at all.
+
```
Comment (bshastry) [inline docs/abi-spec.rst:520]: So this means a non payable function signature should, for example, look like

```
// Non payable
function f() public;

// Payable
function g() public payable;

// View
function h() public view;

// Pure
function i() public pure;
```

Correct?

**Review by bshastry — APPROVED** [review]
LGTM. I'm not sure an example is needed since the definition is clear. I still left a comment just to be sure before I map nonpayable to a state mutability protobuf enum that should simply return an empty string upon conversion to Solidity code.

File: docs/abi-spec.rst (hunk 1 above)
Comment (leonardoalt) [reply docs/abi-spec.rst:520]: Exactly

Comment (bshastry) [reply docs/abi-spec.rst:520]: Just one more thing, library functions may be pure, view, or non payable right? Essentially payable is disallowed for library functions?

**Pull request merged**